### ISSUE TITLE
EIP: 681 - Implement / Adjust URI Scheme for Sending / Receiving TXs

### ISSUE THREAD

**Issue by tayvano** [MyCryptoHQ/MyCrypto#356]
We met up at Devcon to discuss and finalize the proposed specification. As soon as this is implemented on alpha.myetherwallet.com, please let me know. Another wallet (walleth) will be implementing it and if 2 wallets implement it, it will become the standard. As there are so few wallets, the people in the following thread will likely be waiting on us to implement. 

https://github.com/ethereum/EIPs/pull/681/files

**This spec should be used to:**

- [ ] If a user arrives at myetherwallet.com?_____________________________________, it should auto-fill the transaction details for them. 
    - you can see how it currently works here: https://www.myetherwallet.com/?to=0x7cB57B5A97eAbe94205C07890BE4c1aD31E486A8&value=1&tokenSymbol=REP&gaslimit=50000#send-transaction
    - It is not necessary to have the v3 implementation of the URI scheme be supported in v4
    - it used to be that a message popped up at the top before unlocking that would let the user know: `You arrived via a link that provided pre-filled details for the transaction. Please unlock your wallet to send this transaction. You will be able to check and change any of these values before sending`
- [x] A user should be able to generate a QR code to request funds be sent to them (similar to other wallets. This is likely blocked by ongoing discussions surrounding the "View Wallet Info" page / UX / flow
- [ ] Any other QR codes or URI params that we support should be discussed and implemented in a way that makes sense and ensures that someone intending to send a transaction does not get confused for adding a custom node or vise versa. Basically, let's make sure that all URI params are documented somewhere and ensure consistency and that they don't interfere with one another. 

**To Do**

- [ ] Implement on v4
- [ ] Bring up any issues encountered or improvements or concerns
- [ ] Comment on thread
- [ ] Create new issue discussing how we can hijack links (like Gmail does via https://chrome.google.com/webstore/detail/mailto-for-gmail/dgkkmcknielgdhebimdnfahpipajcpjn?hl=en-GB) for people who prefer MyEtherWallet.com.
- [ ] Create a new issue for ensuring future implementations of MEW (e.g. Electron app) comply and are usable and stuff

**Unicode shit**

We should be very mindful of the potential for Unicode scam stuff, especially with ENS names. Read this: https://gojko.net/2017/11/07/five-things-about-unicode.html

Probably smart to write tests and ensure that things are all properly normalized and if a user provides an invalid or scammy parameter, we should inform and educate and do all that fun stuff rather than failing silently or a generic error message. 

**Lastly...**
- Please, do not hesitate to speak directly with @nagydani and @ligi regarding implementation guidance, or comment directly on that thread. This is a group effort and all folks involved are very determined to finally have a spec for this and are happy to answer questions or discuss. 
- If there are any concerns or issues or things that need clarifying during implementation (e.g. things aren't clear, recommendations or guidelines that should be added to spec, etc.) please please please comment on that thread. 
- Feel free to edit or request edits on this issue in order to make it as productive and clear and organized as possible.

**Comment by dternyak** [comment]
Note: this is currently blocked by https://github.com/MyEtherWallet/MyEtherWallet/issues/355

Will be resolved as soon as https://github.com/MyEtherWallet/MyEtherWallet/issues/355 is merged

**Comment by wbobeirne** [comment]
The EIP and this code is in, I'm calling this one closed.
